client.go

```diff
@@ -208,6 +212,8 @@ type Client struct {
 	contentTypeDecoders      map[string]ContentTypeDecoder
 	contentDecompressorKeys  []string
 	contentDecompressors     map[string]ContentDecompressor
+	stopChan                 chan bool
```

Comment:
@cdoucy, I wonder if we need an additional channel to stop the ticker. I will think about it.

Reply:
The other approach I considered is to maintain a slice that would keep track of all `time.Ticker` instances, and the `Close()` method would iterate over this slice and stop all the tickers.

I judged that the channel approach is easier to implement and more in line with golang idiomatics.

Do you see any other approaches?

Reply:
I didn't get a chance to look at it deeper. We can keep the current approach. If needed, I can improve it afterward.